File: Core/DataFilter/CS/DataFilterExplorer/Extensions/AggregateExtension.cs
using DataFilterExplorer.Samples;
using System;
using System.Collections.Generic;
using System.Linq;

namespace DataFilterExplorer
{
    public static class AggregateExtensions
    {
        public static double Aggregate(this IEnumerable<double> values, AggregateFunction type, Func<IEnumerable<double>, double> customFun = null)
        {
            if (customFun != null)
            {
                return customFun(values);
            }
            switch (type)
            {
                case AggregateFunction.Avg:
                    return values.Average();
                case AggregateFunction.Sum:
                    return values.Sum();
                case AggregateFunction.Max:
                    return values.Max();
                case AggregateFunction.Min:
                    return values.Min();
                default:
                    return values.Sum();
            }
        }
    }
}

File: NetFramework/DataFilter/VB/ChartAsDataFilter/Extensions/AggregateExtension.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Runtime.CompilerServices

Namespace ChartAsDataFilter
    Module AggregateExtensions
        <Extension()>
        Function Aggregate(ByVal values As IEnumerable(Of Double), ByVal type As AggregateFunction, ByVal Optional customFun As Func(Of IEnumerable(Of Double), Double) = Nothing) As Double
            If customFun IsNot Nothing Then
                Return customFun(values)
            End If

            Select Case type
                Case AggregateFunction.Avg
                    Return values.Average()
                Case AggregateFunction.Sum
                    Return values.Sum()
                Case AggregateFunction.Max
                    Return values.Max()
                Case AggregateFunction.Min
                    Return values.Min()
                Case Else
                    Return values.Sum()
            End Select
        End Function
    End Module
End Namespace

